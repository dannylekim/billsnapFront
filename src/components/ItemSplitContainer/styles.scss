@import "../../variables";

.confirm__button {
  width: 25%;
  margin: 0 4.5em 2em;
}

.send__right {
  display: flex;
  justify-content: flex-end;
}

//we want to specifically target just this modal
div.popout__button div div .modal-content{
  width: 85% !important;
}

.popout__button {
  line-height: 1em;
  margin-left: 2px;
  opacity: 0.6 !important;
  transition: 0.5s ease;
  cursor: pointer;
}

.popout__button:hover{
  opacity:1 !important;
  transition: 0.5s ease;
}

.scroll__bar__modal {
  overflow-y: auto;
  height: 80%;
}

.scroll__bar__modal::-webkit-scrollbar {
  width: 0;
}