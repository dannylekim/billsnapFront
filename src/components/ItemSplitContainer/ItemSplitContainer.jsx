import React, {Component} from "react";
import {MdOpenInNew} from "react-icons/md";

import "./styles.scss";
import ItemSplit from "../ItemSplit";
import {Button, Modal} from "shards-react";

function getSelectedItem(info, selectedItemId) {
  return info.items.find((item) => item.itemId === selectedItemId);
}

class ItemSplitContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      itemInformation: {},
    };

    this.toggleModal = this.toggleModal.bind(this);
  }

  componentDidMount() {
    const { selectedItemId, bill } = this.props;
    const selectedItem = bill.items.find((item) => item.id === selectedItemId);

    const itemInformation = {
      name: selectedItem.name,
      cost: selectedItem.cost,
      accounts: [],
    };

    bill.informationPerAccount
      .filter((info) => {
        return getSelectedItem(info, selectedItemId);
      })
      .forEach((involvedInformationPerAccount) => {
        itemInformation.accounts.push({
          status: involvedInformationPerAccount.invitationStatus,
          firstName: involvedInformationPerAccount.account.firstName,
          percentage: getSelectedItem(
            involvedInformationPerAccount,
            selectedItemId
          ).percentage,
        });
      });

    this.setState((prev) => ({
      ...prev,
      itemInformation: itemInformation,
    }));
  }

  toggleModal() {
    this.setState((prev) => ({
      ...prev,
      isOpen: !this.state.isOpen,
    }));
  }

  render() {
    const { isOpen, itemInformation } = this.state;

    return (
      <div className="popout__button">
        <MdOpenInNew onClick={this.toggleModal} size={15} />
        <Modal
          open={isOpen}
          toggle={this.toggleModal}
          className="scroll__bar__modal"
        >
          <ItemSplit itemInformation={itemInformation} />
          <div className="send__right">
            <Button
              pill
              size="sm"
              theme="success"
              className="confirm__button"
              onClick={this.toggleModal}
            >
              Confirm
            </Button>
          </div>
        </Modal>
      </div>
    );
  }
}

export default ItemSplitContainer;
