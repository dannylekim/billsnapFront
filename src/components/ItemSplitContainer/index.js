import ItemSplitContainer from "./ItemSplitContainer";

export default ItemSplitContainer;