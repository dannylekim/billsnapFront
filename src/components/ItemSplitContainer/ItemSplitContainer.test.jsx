import React from "react";
import ItemSplitContainer from "./ItemSplitContainer";

const testInformation = {
  name: "Bigger String",
  cost: "300.00",
  accounts: [
    { status: "ACCEPTED", firstName: "string", percentage: 60 },
    { status: "ACCEPTED", firstName: "string", percentage: 40 },
  ],
};

describe("ItemSplitContainer", () => {
  describe("render", () => {
    const selectedItemId = 52;

    describe("snapshot ðŸ“¸", () => {
      it("should match closed snapshot", () => {
        matches(
          <ItemSplitContainer
            selectedItemId={selectedItemId}
            itemInformation={testInformation}
            setActiveItemId={jest.fn}
          />
        );
      });

      it("should match open snapshot", () => {
        const wrapper = shallow(
          <ItemSplitContainer
            selectedItemId={selectedItemId}
            itemInformation={testInformation}
            setActiveItemId={jest.fn}
          />
        );

        wrapper.setState((prev) => ({
          ...prev,
          isOpen: true,
        }));

        matches(wrapper);
      });
    });
  });
});
