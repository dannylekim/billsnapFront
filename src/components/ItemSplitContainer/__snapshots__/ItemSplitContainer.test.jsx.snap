// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSplitContainer render snapshot ðŸ“¸ should match closed snapshot 1`] = `
<div
  className="popout__button"
>
  <svg
    fill="currentColor"
    height={15}
    onClick={[Function]}
    size={15}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={15}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
    />
  </svg>
</div>
`;

exports[`ItemSplitContainer render snapshot ðŸ“¸ should match open snapshot 1`] = `
<div
  className="popout__button"
>
  <svg
    fill="currentColor"
    height={15}
    onClick={[Function]}
    size={15}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={15}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
    />
  </svg>
  <div
    className="modal-backdrop fade"
  />
  <div
    aria-hidden={true}
    className="modal fade top"
    onClick={[Function]}
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog scroll__bar__modal"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal__text modal__body card-body"
        >
          <div>
            Item: 
            <span
              className="embolden"
            >
               
              Bigger String
               
            </span>
          </div>
          <div>
            Total: 
            300.00
             $
          </div>
          <hr
            className="card__separator"
          />
          <div>
            Shared between:
          </div>
          <div
            className="one__liner"
          >
            <div>
              <div
                className="accounts__icons"
              >
                <svg
                  fill="currentColor"
                  height={50}
                  size={50}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={50}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
                <svg
                  className="status_icon"
                  color="#47E5B6"
                  fill="currentColor"
                  height={40}
                  size={40}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "#47E5B6",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={40}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  />
                </svg>
              </div>
              <p
                className="icon__name"
              >
                string
              </p>
            </div>
            <div
              className="grow__line"
            >
              <div
                className="center"
              >
                180.00
                 $
              </div>
              <div
                className="progress"
              >
                <div
                  aria-valuemax={100}
                  aria-valuemin="0"
                  aria-valuenow={60}
                  className="progress-bar bg-success"
                  role="progressbar"
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                >
                  60
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="one__liner"
          >
            <div>
              <div
                className="accounts__icons"
              >
                <svg
                  fill="currentColor"
                  height={50}
                  size={50}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width={50}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
                <svg
                  className="status_icon"
                  color="#47E5B6"
                  fill="currentColor"
                  height={40}
                  size={40}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "#47E5B6",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={40}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  />
                </svg>
              </div>
              <p
                className="icon__name"
              >
                string
              </p>
            </div>
            <div
              className="grow__line"
            >
              <div
                className="center"
              >
                120.00
                 $
              </div>
              <div
                className="progress"
              >
                <div
                  aria-valuemax={100}
                  aria-valuemin="0"
                  aria-valuenow={40}
                  className="progress-bar bg-success"
                  role="progressbar"
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                >
                  40
                   %
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="send__right"
        >
          <button
            className="confirm__button btn btn-success btn-sm btn-pill"
            onClick={[Function]}
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
